libraries/provider_rightscale_volume.rb

```diff
@@ -556,7 +556,15 @@ def attach_volume(volume_id, device)
         scan_for_attachments if node['virtualization']['system'] == 'vmware'
 
         # Determine the actual device name
-        (Set.new(get_current_devices) - current_devices).first
+        added_device = nil
+        Timeout::timeout(60) do
+          begin
+            sleep(1)
+            Chef::Log.info "Checking for added device."
+            added_device = (Set.new(get_current_devices) - current_devices).first
+          end until !added_device.nil?
+        end
```

Comment:
The condition needs to be determined within the loop so it would make sense to set a condition after it has been evaluated.

And @douglaswth is correct that best to replace the double negative (! nil) with `end until added_device`.